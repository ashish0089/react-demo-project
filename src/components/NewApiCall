import React from 'react';


export default class getExpense extends React.Component {
  constructor(props) {
    super(props);
    this.onTextChange = this.onTextChange.bind(this);

    this.state = {
      text: '',
      getJson: ''
    };
  }

  onTextChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({ note }));
    if(note.length>3){
        fetch(' ')
  .then(function(response) {
    return response.json();
  })
  .then(function(myJson) {
   this.setState(() => ({ getJson:myJson }));
  });
    }
  };

  
  render() {
    return (
      <div>
  
          <input
            type="text"
            placeholder="text"
            value={this.state.text}
            onChange={this.onTextChange}
          />
          
          
        </form>
        <ChildComponet key ={getJson.id} listItem = {getJson}/>
      </div>
    )
  }
}
