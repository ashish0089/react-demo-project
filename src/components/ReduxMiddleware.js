import { createStore, applyMiddleware } from 'redux';


const initialState = {
    count:0
}
const reducerFunction = (state, action) =>{
    switch(action.type){
        case "INC": 
            return state+1;
        
        case "DEC": 
            return state-1;
        default:
        return state;
    }

}

const middleWare = applyMiddleware();

const store = createStore(reducerFunction, applyMiddleware);
