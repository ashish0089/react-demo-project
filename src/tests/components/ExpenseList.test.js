import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import Expenses from '../fixtures/expenses';


test('this is used to test list with expense data',() =>{
    const Wrapper = shallow(<ExpenseList expenses ={Expenses}/>);
    expect(Wrapper).toMatchSnapshot();
});

test('this is used to test expense list with no data', () =>{
    const Wrapper = shallow(<ExpenseList expenses ={[]}/>);
    expect(Wrapper).toMatchSnapshot();
});