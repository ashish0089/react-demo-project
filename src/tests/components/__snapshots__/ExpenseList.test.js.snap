// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`this is used to test expense list with no data 1`] = `
<div>
  <div>
    no expense data
  </div>
</div>
`;

exports[`this is used to test list with expense data 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={-345600000}
    description="Rent"
    id="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit Card"
    id="3"
    note=""
  />
</div>
`;
